__tests__/app.test.js

```diff
@@ -132,22 +123,61 @@ describe("GET /api/reviews/:review_id/comments", () => {
 				expect(body.comments).toEqual([]);
 			});
 	});
-  
+
 	test("404 Not found if review_id isn't in the database", () => {
 		return request(app)
 			.get("/api/reviews/78/comments")
 			.expect(404)
-			.then(({ body }) => {
-				expect(body).toEqual({ msg: "Not found" });
-			});
+			.then(({ body }) => expect(body).toEqual({ msg: "Not found" }));
 	});
 	test("400 If key is not the correct data type", () => {
 		return request(app)
 			.get("/api/reviews/hello/comments")
 			.expect(400)
+			.then(({ body }) => expect(body).toEqual({ msg: "Bad Request" }));
+	});
+});
+
+describe("POST /api/reviews/:review_id/comments should post a new comment when given a body", () => {
+	test("201 should add a new comment to the comments database", () => {
+		return request(app)
+			.post("/api/reviews/2/comments")
+			.send({ author: "mallionaire", body: "Great for beginners" })
+			.expect(201)
 			.then(({ body }) => {
-				expect(body).toEqual({ msg: "Bad Request" });
+				expect(body.comment[0]).toHaveProperty("comment_id");
+				expect(body.comment[0]).toHaveProperty("body");
+				expect(body.comment[0]).toHaveProperty("review_id");
+				expect(body.comment[0]).toHaveProperty("author");
+				expect(body.comment[0]).toHaveProperty("created_at");
+				expect(body.comment[0].review_id).toEqual(2);
+				expect(body.comment[0].author).toEqual("mallionaire");
+				expect(body.comment[0].body).toEqual("Great for beginners");
```

Comment:
I think your response body should have a key of `comment` which is an object, rather than an array containing just a single comment object.

Also, the way you have asserted for each property individually I really like!
If you are asserting the value of a property, you don't need to also check that is exists. Keep in mind the most appropriate matcher for the task - `.toBe` would be preferable over `.toEqual` here

Reply:
thanks, i did wonder about the redundancy of checking the property value and the toBe values of the same thing, but I did both just to be safe. I'll change it now

toBe vs toEqual is something I need to learn more about, thanks for mentioning it